plugins/modules/katello_content_view_version.py

```diff
@@ -143,58 +127,34 @@
     state: absent
 '''
 
-RETURN = '''# '''
+RETURN = ''' # '''
 
-try:
-    from nailgun.entities import (
-        ContentViewVersion,
-    )
 
-    from ansible.module_utils.ansible_nailgun_cement import (
-        create_server,
-        find_organization,
-        find_lifecycle_environments,
-        find_lifecycle_environment,
-        find_content_view,
-        find_content_view_version,
-        ping_server,
-        set_task_timeout,
-        naildown_entity_state,
-    )
-    has_import_error = False
-except ImportError as e:
-    has_import_error = True
-    import_error_msg = str(e)
-from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.foreman_helper import KatelloEntityAnsibleModule
 
 
-def promote_content_view_version(module, content_view_version, organization, environments, synchronous, **kwargs):
+def promote_content_view_version(module, content_view_version, environments, synchronous, force, force_yum_metadata_regeneration):
     changed = False
 
-    current_environment_ids = map(lambda environment: environment.id, content_view_version.environment)
-    desired_environment_ids = map(lambda environment: environment.id, environments)
-    promote_to_environment_ids = list(set(desired_environment_ids) - set(current_environment_ids))
-
-    request_data = {'environment_ids': promote_to_environment_ids}
-    request_data.update({k: v for k, v in kwargs.items() if v is not None})
+    current_environment_ids = {environment['id'] for environment in content_view_version['environments']}
+    desired_environment_ids = {environment['id'] for environment in environments}
+    promote_to_environment_ids = list(desired_environment_ids - current_environment_ids)
 
     if promote_to_environment_ids:
-        if not module.check_mode:
-            content_view_version.promote(synchronous, data=request_data)
-        changed = True
+        payload = {
+            'environment_ids': promote_to_environment_ids,
+            'force': force,
+            'force_yum_metadata_regeneration': force_yum_metadata_regeneration,
+        }
+
+        changed, _ = module.resource_action('content_view_versions', 'promote', params=payload)
     return changed
 
 
 def main():
-    module = AnsibleModule(
-        argument_spec=dict(
-            server_url=dict(required=True),
-            username=dict(required=True, no_log=True),
-            password=dict(required=True, no_log=True),
-            verify_ssl=dict(type='bool', default=True),
-            content_view=dict(required=True),
-            organization=dict(required=True),
-            state=dict(default='present', choices=['present', 'absent']),
+    module = KatelloEntityAnsibleModule(
+        entity_spec=dict(
+            content_view=dict(type='entity', flat_name='content_view_id', required=True),
             version=dict(),
             lifecycle_environments=dict(type='list', default=['Library']),
```

Comment:
This contradicts the doc, which lists force_promote as the primary name

Reply:
I will change it to adhere the docs, because in the context of the whole module force is not that clear.